web/src/components/charts/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -330,6 +330,13 @@ const TimeSeriesChart: React.FC<TimeSeriesLinesProps> = ({
             newSelection[key] = key === name;
           });
           chartOptions.legend.selected = newSelection;
+
+          // This checks if everything is going to deselected, if yes we enable all series
+        } else if (Object.keys(selected).every(key => !selected[key])) {
```

Comment:
```suggestion
        } else if (!Object.keys(selected).some(key => selected[key])) {
```

optional mod for readability

Reply:
Updated!